include/deal.II/grid/grid_tools.h

```diff
@@ -158,14 +158,16 @@ namespace GridTools
    */
   template <int dim, int spacedim>
   double
-  minimal_cell_diameter (const Triangulation<dim, spacedim> &triangulation);
+  minimal_cell_diameter (const Triangulation<dim, spacedim> &triangulation,
+                         const Mapping<dim,spacedim> &mapping = (StaticMappingQ1<dim,spacedim>::mapping));
 
   /**
    * Return the diameter of the largest active cell of a triangulation.
    */
   template <int dim, int spacedim>
   double
-  maximal_cell_diameter (const Triangulation<dim, spacedim> &triangulation);
+  maximal_cell_diameter (const Triangulation<dim, spacedim> &triangulation,
+                         const Mapping<dim,spacedim> &mapping = (StaticMappingQ1<dim,spacedim>::mapping));
```

Comment:
I think you should document for these functions that the diameter is computed only by looking at the distances between vertices. For very deformed cells, this is not the diameter of the area/volume occupied by this cell.

Reply:
I have not changed the original implementation. That's exactly what the function was doing before (i.e., that's what cell->diameter does). I'll document this as well.

Reply:
BTW, the comments you made should really be on #6295. This branch is based on that one. I'll rebase afterwards.